qdb/comm.py

```diff
@@ -366,7 +366,8 @@ def user_stop(self):
         """
         if self.reader and self.reader.is_alive():
             self.reader.terminate()
-        self.socket.close()
+        if self.socket is not None:
```

Comment:
with the changes to the `__new__`, I don't think we need these extra None checks. I am not sure we can hit these conditions anymore.